src/link/section.cpp

```diff
@@ -2,35 +2,23 @@
 
 #include <assert.h>
 #include <inttypes.h>
+#include <map>
 #include <stdlib.h>
+#include <string>
 #include <string.h>
 
 #include "link/main.hpp"
 #include "link/section.hpp"
 
 #include "error.hpp"
-#include "hashmap.hpp"
 #include "linkdefs.hpp"
 
-HashMap sections;
+std::map<std::string, struct Section *> sections;
 
-struct ForEachSectionArg {
-	void (*callback)(struct Section *section, void *arg);
-	void *arg;
-};
-
-static void forEach(void *section, void *arg)
+void sect_ForEach(void (*callback)(struct Section *))
 {
-	struct ForEachSectionArg *callbackArg = (struct ForEachSectionArg *)arg;
-
-	callbackArg->callback((struct Section *)section, callbackArg->arg);
-}
-
-void sect_ForEach(void (*callback)(struct Section *, void *), void *arg)
-{
-	struct ForEachSectionArg callbackArg = { .callback = callback, .arg = arg};
-
-	hash_ForEach(sections, forEach, &callbackArg);
+	for (auto &it : sections)
```

Comment:
How about
```suggestion
	for (auto &[name, section] : sections)
```
?

Reply:
`name` would be an unused variable; is there a solution for that? `for (auto &[, section] : sections)`?